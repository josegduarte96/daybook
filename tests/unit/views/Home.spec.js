import {shallowMount} from '@vue/test-utils'
import Home from '@/views/Home'


describe('Pruebas en Home view', () => {

    test('El componente debe renderizar correctamente', () => {
        const wrapper = shallowMount( Home )
        expect(wrapper.html()).toMatchSnapshot()
    })

    test('Hacer click en un boton debe redireccionar a una ruta no-entry ', () => {
        const mockRouter = {
            push: jest.fn()
        }

        const wrapper = shallowMount(Home, {
            global: {
                mocks: {
                    $router: mockRouter
                }
            }
        })

        wrapper.find('button').trigger('click')

        expect(mockRouter.push).toHaveBeenCalled()
    })
    
})