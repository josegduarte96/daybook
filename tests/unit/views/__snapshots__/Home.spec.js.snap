// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en Home view El componente debe renderizar correctamente 1`] = `<div class="home"><img alt="Vue logo" src=""><button class="btn btn-dark">Primary</button></div>`;
